### PR TITLE
Support file extension in the RABBITMQ_CONFIG_FILE env var values

### LINKED ISSUE #691: Support config file(s) path values with and without file extensions
Currently we add the `.config`/`.conf` (once #550 is merged) suffix to `RABBITMQ_CONFIG_FILE` before attempting to do anything with it. This is fairly counter-intuitive and makes some of our documentation examples look confusing. It would be a lot more user friendly to support values with and without file extensions.

### PR DESCRIPTION
**Pull request by hairyhum**
Fixes #691

### CODE REVIEW

**Comment by michaelklishin** [comment]
This breaks support for paths without a filename extension. The problematic line seems to be
the one with `"${RABBITMQ_CONFIG_FILE%.*}"`:

```
export RABBITMQ_CONFIG_FILE=./etc/rabbitmq/rabbitmq
echo "${RABBITMQ_CONFIG_FILE%.*}"

export RABBITMQ_CONFIG_FILE=./etc/rabbitmq/rabbitmq.config
echo "${RABBITMQ_CONFIG_FILE%.*}"
./etc/rabbitmq/rabbitmq
export RABBITMQ_CONFIG_FILE=./etc/rabbitmq/rabbitmq.sysctl
echo "${RABBITMQ_CONFIG_FILE%.*}"
./etc/rabbitmq/rabbitmq
export RABBITMQ_CONFIG_FILE=./etc/rabbitmq/rabbitmq.sysctl.conf
echo "${RABBITMQ_CONFIG_FILE%.*}"
./etc/rabbitmq/rabbitmq.sysctl
```

**Comment by michaelklishin** [comment]
I've pushed something that fixes this particular problem by using `dirname` and `basename`. I'm hardly great at shell script, so @hairyhum and @dumbbell, please double check.

We also need to work on supporting both `advanced` and `advanced.config`.

**Comment by hairyhum** [comment]
Or we can just change conditions, because extension is being checked there anyway. So we can use any extension trimming mechanism.

**Comment by michaelklishin** [comment]
@hairyhum this passes the acceptance criteria for `rabbitmq.conf` and `rabbitmq.config` but not `advanced.config`. I think we should support both options for the `advanced.config` as well.

**Comment by hairyhum** [comment]
Added support for `advanced.config`. Haven't checked windows scripts yet.

**Comment by michaelklishin** [comment]
This looks good on OS X (zsh) and Ubuntu (dash). Haven't had a chance to test on Windows yet.

File: scripts/rabbitmq-service.bat [hunk 1]
```diff
@@ -171,8 +178,27 @@ if exist "!RABBITMQ_CONFIG_FILE!.config" (
                                 -conf_advanced "!RABBITMQ_ADVANCED_CONFIG_FILE!" ^
                                 -config "!RABBITMQ_ADVANCED_CONFIG_FILE!"
     )
+) else (
+    if exist "!RABBITMQ_CONFIG_FILE!.config" (
+        set RABBITMQ_CONFIG_ARG=-config "!RABBITMQ_CONFIG_FILE!"
+    ) else if exist "!RABBITMQ_CONFIG_FILE!.config" (
```
Comment (Gsantomaggio) [inline scripts/rabbitmq-service.bat]: this `else if`  should be `if exist "!RABBITMQ_CONFIG_FILE!.conf"` as [`rabbitmq-server.bat`](https://github.com/rabbitmq/rabbitmq-server/blob/rabbitmq-server-691/scripts/rabbitmq-server.bat#L93)

File: scripts/rabbitmq-service.bat [hunk 2]
```diff
@@ -126,6 +126,11 @@ set RABBITMQ_EBIN_ROOT=!RABBITMQ_HOME!\ebin
 
 set RABBITMQ_CONFIG_FILE="!RABBITMQ_CONFIG_FILE!"
```
Comment (Gsantomaggio) [inline scripts/rabbitmq-service.bat]: I am not sure this `set` is necessary. 
In [`rabbitmq-server.bat`](https://github.com/rabbitmq/rabbitmq-server/blob/rabbitmq-server-691/scripts/rabbitmq-server.bat#L43) is not present.

**Pull request merged**